server/http/controller.go

```diff
@@ -280,10 +281,50 @@ func (c *controller) UpdateTest(ctx context.Context, testID string, in openapi.T
 	updated.ID = test.ID
 	updated.ReferenceRun = nil
 
+	testHasChanged, err := c.testHasChanged(test, updated)
```

Comment:
all this logic should live outside the controller. I'm thinking we could have it in `server/model/test_changes.go` and have somehting like:

```go
func ChangeTest(in, updated Test) (Test, error) {
  testHasChanged, err := testHasChanged(test, updated)
  if err != nil {
    return Test{}, err
  }

  if testHasChanged {
    updated.Version = test.Version + 1
  }

  return updated
}
```

and move all the `testHasChanged` etc methods there too

Reply:
Great idea. I just pushed the change